### ISSUE TITLE
Pre-trained model and audio samples.

### ISSUE THREAD

**Issue by Rayhane-mamah** [Rayhane-mamah/Tacotron-2#30]
Hello everyone, just wanted to drop some naturally synthesized audio samples (griffin-lim inversion of mels) along with the [model](https://drive.google.com/drive/folders/1DoddWGKce1mxzry4I36Fe0Ys05OHVckg) that generated them. Enjoy!

[samples.tar.gz](https://github.com/Rayhane-mamah/Tacotron-2/files/1944094/samples.tar.gz)
 
**Sentences:**
- hello everyone and thank you for hearing me out!
- these are probably my first words as a pre-trained model.
- finally, today, after about 4 months of development I am able to speak naturally
- my voice is still robotic indeed, that\'s why my developer is working on Wavenet vocoder at this particular moment.
- I still have some small problems with caps, words context and typos. but I read stuff differently each time.
- Let me show you what I mean!
- even if it doesn't sound like it, I do have feelings!
- even if it doesn't sound like it, I do have feelings! (repetition to show slight difference in generation)
- this is due to the pre-net drop-out even at inference time.
- another thing I want to mention is that I am currently able to read long sentences with ease. here\'s an example!
- Sequence to sequence models have enjoyed great success in a variety of tasks such as machine translation, speech recognition, and text summarization. This tutorial gives readers a full understanding of sequence to sequence models and shows how to build a competitive sequence to sequence model from scratch. We focus on the task of neural machine translation which was the very first wild success for this architecture.
- the tutorial should be in the git-hub comment.  ( [this one](https://www.tensorflow.org/tutorials/seq2seq) )
- you also probably noticed how well I can read complex words.
- so to wrap things up, I am slowly progressing, and hopefully I will speak like humans very soon.
- Thank you all so much for your support!

**Comment by gloriouskilka** [comment]
Could you write your hparams, if they differ from default, please? I'd like to try to retrain this model on different data

**Comment by Rayhane-mamah** [comment]
The hparams used for this model are the exact same parameters in [hparams.py](https://github.com/Rayhane-mamah/Tacotron-2/blob/master/hparams.py) (Default).

- You could however try reducing fmin to 0 ([reason](https://github.com/Rayhane-mamah/Tacotron-2/issues/33))
- You can also play a bit with [learning rate decay parameters](https://github.com/Rayhane-mamah/Tacotron-2/blob/master/hparams.py#L103) to make learning rate drop faster or slower which might have a tiny impact as well. Our decay is based on the [exponential decay](https://www.tensorflow.org/api_docs/python/tf/train/exponential_decay) so you can pretty much estimate the lr schedule in advance.

It appears that a model with outputs_per_step=5 will make things harder for Wavenet vocoder, so I will think about training some model with outputs_per_step=2 or 3 for later experiments. Quality is important indeed, but speed is also a priority..

**Comment by MXGray** [comment]
@Rayhane-mamah
Awesome work! Can't wait for wavenet vocoder training and synthesis to be implemented. :)

**Comment by candlewill** [comment]
Awesome work! @Rayhane-mamah Do you think that the speed of the generated voice is faster than the original voice? It exists in my experiment, and I don't know why.

**Comment by Rayhane-mamah** [comment]
Thanks @candlewill, I do think it goes faster than ground truth.

Someone mentionned that using zoneout at inference too causes the speed to become more or less the same. It has been mentionned [here](https://github.com/Rayhane-mamah/Tacotron-2/issues/4#issuecomment-384894124)

I did however notice that quality tends to become worse when using zoneout at inference times. I also noticed that further training improves speed slightly.

I will look into it further and tell you how it goes.

**Comment by PetrochukM** [comment]
@Rayhane-mamah How many steps did you run the training for? What was the loss at the end?

**Comment by MXGray** [comment]
@Rayhane-mamah - No news yet regarding wavenet training and synthesis? :) The suspense is killing me - Haha! ;)

**Comment by Rayhane-mamah** [comment]
@MXGray haha I am so sorry for this delay :)

Actually the implementation is pretty much complete, I am just debugging and fixing stuff here and there (mainly due to people's requests). I am also caught in an exams week so things are a little messy right now.. 

Sorry again for the delay, I will try my best to release it as soon as it's ready! :)

**Comment by stevemurr** [comment]
@MXGray In case you're interested, I was able to successfully use this Tacotron 2 implementation in concert with r9y9's repo achieving comparable results to the samples in Google's publication.  I tried training the WaveNet on both ground truth and mel spectrogram predictions however if you train the WaveNet long enough, using ground truth is actually not an issue.  I trained the WaveNet to approximately 2 million steps and the Tacotron model to roughly 250k steps.   

If you choose to try this I can share my hparams for each model.  The key being when synthesizing the mel spectrogram you'll need to move the range from `[-4, 4]` to `[-1, 1]` (this target range will depend on your WaveNet training settings) with something like:

`x = np.interp(x, (-4, 4), (-1, 1))`

In one of the recent commits to this repo, the mel axes were swapped (should be (N, 80) for r9y9), so if that's still the case you will need to swap them using something like:

```
if x.shape[1] != 80:
    np.swapaxes(x, 0, 1)
```

At which point, the WaveNet will work as expected :).

**Comment by PetrochukM** [comment]
@stevemurr Can you share samples?

**Comment by r9y9** [comment]
@stevemurr That's great! I'm glad to hear that you used my repo and got comparable results to the samples in Google's. I'm curious what dataset are you working on? Also, could you share samples?

**Comment by fatchord** [comment]
@Rayhane-mamah Excellent results, congrats!

@stevemurr Would love to hear samples of your taco->wavenet!

**Comment by neverjoe** [comment]
I test my way described in https://github.com/Rayhane-mamah/Tacotron-2/issues/29 with @stevemurr's, [wavenet vocoder](https://drive.google.com/file/d/1tptZDDZ8rOWUdse0HWn9W2cm0yN0xhx6/edit)
and taco2 was trained round 500k. 
i think @stevemurr's way is better.
[sample-taco2-wavenet.zip](https://github.com/Rayhane-mamah/Tacotron-2/files/1974349/sample-taco2-wavenet.zip)

**Comment by Rayhane-mamah** [comment]
@stevemurr congrats for your results, I'm glad my work helped reaching results comparable to google's paper! :)

Do you mind sharing your models hparams? Thanks a lot.

**Comment by stevemurr** [comment]
@Rayhane-mamah Of course!  These are the `hparams` I override for the Tacotron model:
```
{
    "sample_rate": 24000,
    "power": 1.1,
    "input_type": "raw",
    "quantize_channels": 65536,
    "outputs_per_step": 3 
}
```
Shoot me an email at `stevemurr@gmail.com` and I can share you some examples.

**Comment by butterl** [comment]
@stevemurr  Thanks for your sharing！ Could you also share the hparams that you override in wavenet of r9y9's repo?

**Comment by zhf459** [comment]
@stevemurr  great to hear that you get good result ,can you share a pre-trained wavenet model?

**Comment by r9y9** [comment]
Here are samples of Tacotron2 + WaveNet trained on LJSpeech: [taco2.zip](https://github.com/Rayhane-mamah/Tacotron-2/files/1988903/taco2.zip) . I think it's pretty good. I plan to release pretraind models.

**Comment by begeekmyfriend** [comment]
![wavenet](https://user-images.githubusercontent.com/6031938/39848076-8a028bb8-5437-11e8-901f-95a7a4388d3a.png)
![linear](https://user-images.githubusercontent.com/6031938/39848077-8d170176-5437-11e8-88a1-d493bec95b74.png)
Wow, so expecting... The images of spectrogram from wavenet look much better than that from linear spectrograms

**Comment by r9y9** [comment]
Uploaded more samples at https://r9y9.github.io/wavenet_vocoder/. You can find pretrained models and hyper parameters on the page. I fine-turned the pretrained model provided here by @Rayhane-mamah. It's really great. Let me say thank you very much.

**Comment by Rayhane-mamah** [comment]
@r9y9, Awesome work on combining the two model for the final Tacotron-2.

There is something I would like to propose in private, if you're interested, shoot me an email at `rayhane.mamah@gmail.com`.

Thank you for the samples and pre trained models! :)

**Comment by begeekmyfriend** [comment]
@r9y9 Do these PRs complete the final Tacotron2? Brovo!!

**Comment by MXGray** [comment]
@r9y9
Cool! Would greatly appreciate if you could share your combined end-to-end implementation of Tacotron2 by @Rayhane-mamah and your WaveNet VoCoder.  :)

**Comment by alex73** [comment]
I tried to train LJSpeech-1.1 model myself by the "python3 train.py --model='Tacotron'" command, then  synthesize eval texts, but quality is worse than in samples.tar.gz. I made only one change in the hparams.py: I changed tacotron_batch_size into 32, because 1080 GPU doesn't have enough memory.
File from samples.tar.gz: https://drive.google.com/file/d/1u7T4wBmoyRUIm1tyrCqvbdXOL_6QBHn9/view?usp=sharing
My file: https://drive.google.com/file/d/1O2luexOp9fPLCu0U0ZOjUsFIQzdjhSQ4/view?usp=sharing
What is wrong ?

**Comment by Rayhane-mamah** [comment]
You may want to train for more global steps. Also, if you don't plan on using Wavenet, set "predict_linear" to true in hparams for better audio quality.

**Comment by mrgloom** [comment]
Should this pretrained model work with current master?
